src/uptane/tuf.cc

```diff
@@ -91,12 +91,9 @@ bool Hash::MatchWith(const std::string &content) const {
 
 bool Hash::operator==(const Hash &other) const { return type_ == other.type_ && hash_ == other.hash_; }
 
-Target::Target(const std::string &filename, const Json::Value &content) : filename_(filename), ecu_identifier_("") {
+Target::Target(const std::string &filename, const Json::Value &content) : filename_(filename) {
   if (content.isMember("custom")) {
-    Json::Value custom = content["custom"];
-    if (custom.isMember("ecuIdentifier")) {
-      ecu_identifier_ = content["custom"]["ecuIdentifier"].asString();
-    }
+    custom_ = content["custom"];
```

Comment:
Why is this needed?  I'm not keen on exposing implementation details (i.e. the format of the 'custom' JSON field, or the fact that the serialization of Targets happens to be JSON at the moment) via the API of Uptane::Target.

What didn't work with the old method?